pkg/urbit/jets/e/en_json.c

```diff
@@ -0,0 +1,328 @@
+/* j/5/en_json.c
+**
+*/
+#include "all.h"
+
+/* helper macros
+*/
+
+#ifndef SERIALIZE_BUFFER_INIT_SIZE
+  #define SERIALIZE_BUFFER_INIT_SIZE 4096
+#endif
+
+#ifndef SERIALIZE_BUFFER_INCR_SIZE
+  #define SERIALIZE_BUFFER_INCR_SIZE 4096
+#endif
+
+/* helper structs
+*/
+
+typedef struct _json_buffer {
+  c3_y *buf_y;
+  c3_w len_w;
+  c3_w siz_w;
+} json_buffer;
+
+/* constants
+*/
+
+const c3_y _JSON_NULL[] = "null";
+const c3_y _JSON_TRUE[] = "true";
+const c3_y _JSON_FALSE[] = "false";
+const c3_y _JSON_NEWLINE[] = "\\n";
+const c3_y _JSON_DOQ[] = "\\\"";
+const c3_y _JSON_BAS[] = "\\\\";
+const c3_y _JSON_DEL[] = "\\u007f";
+const c3_y *_JSON_UNICODES[] = {
+  "\\u0000",  // 0
+  "\\u0001",  // 1
+  "\\u0002",  // 2
+  "\\u0003",  // 3
+  "\\u0004",  // 4
+  "\\u0005",  // 5
+  "\\u0006",  // 6
+  "\\u0007",  // 7
+  "\\u0008",  // 8
+  "\\u0009",  // 9
+  "\\u000a",  // 10
+  "\\u000b",  // 11
+  "\\u000c",  // 12
+  "\\u000d",  // 13
+  "\\u000e",  // 14
+  "\\u000f",  // 15
+  "\\u0010",  // 16
+  "\\u0011",  // 17
+  "\\u0012",  // 18
+  "\\u0013",  // 19
+  "\\u0014",  // 20
+  "\\u0015",  // 21
+  "\\u0016",  // 22
+  "\\u0017",  // 23
+  "\\u0018",  // 24
+  "\\u0019",  // 25
+  "\\u001a",  // 26
+  "\\u001b",  // 27
+  "\\u001c",  // 28
+  "\\u001d",  // 29
+  "\\u001e",  // 30
+  "\\u001f",  // 31
+};
+
+/* forward declarations
+*/
+
+void
+_serialize(json_buffer*, u3_noun);
+
+/* core jet logic
+*/
+
+void
```

Comment:
All the functions in this file except for the jet interface at the bottom should be declared `static` (in this case, `static void`).

Reply:
Done